pkg/operator/kit/util.go

```diff
@@ -0,0 +1,81 @@
+/*
```

Comment:
as discussed, i would split the functionality into more meaningful files, such as retry.go and rest.go